src/Symfony/Bundle/FrameworkBundle/Debug/TraceableEventDispatcher.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Debug\CliRequest;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher as BaseTraceableEventDispatcher;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class TraceableEventDispatcher extends BaseTraceableEventDispatcher
+{
+    public function __construct(
+        EventDispatcherInterface $dispatcher,
+        Stopwatch $stopwatch,
+        LoggerInterface $logger = null,
+        private readonly ?RequestStack $requestStack = null
+    ) {
+        parent::__construct($dispatcher, $stopwatch, $logger, $requestStack);
+    }
+
+    /**
+     * Starts a stopwatch section before a command runs to profile listeners.
+     */
+    protected function beforeDispatch(string $eventName, object $event): void
+    {
+        if (!$event instanceof ConsoleCommandEvent) {
+            parent::beforeDispatch($eventName, $event);
+
+            return;
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+
+        if (!$request instanceof CliRequest || $request->command !== $event->getCommand()) {
+            return;
+        }
+
+        $request->attributes->set('_stopwatch_token', substr(hash('sha256', uniqid(mt_rand(), true)), 0, 6));
+        $this->stopwatch->openSection(); // section is closed on console.terminate by ConsoleProfilerListener
```

Comment:
having a different class responsible for opening and closing the section is a good way to create broken situations where we don't do both.

Why isn't ConsoleProfilerListener responsible for opening the section on `console.command` ?

Reply:
Ref https://github.com/symfony/symfony/pull/47416#discussion_r1328972985.
Moved to the ConsoleProfilerListener anyway.